commands.js

```diff
@@ -1782,8 +1782,17 @@ var commands = exports.commands = {
 
 	addplayer: function (target, room, user) {
 		if (!target) return this.parse('/help addplayer');
-
-		return this.parse('/roomplayer ' + target);
+		target = this.splitTarget(target);
+		var targetUser = this.targetUser;
+		if (!targetUser) return this.sendReply("User " + this.targetUsername + " not found.");
+		if (!room.joinbattle) return this.sendReply("You can only do this in battle rooms.");
+		if (user.can('joinbattle', null, room) && this.can('hideroom') && !targetUser.can('joinbattle', null, room)) {
```

Comment:
The command that is used in battle rooms is <code>/hideroom</code>, but the permission is still <code>privateroom</code> (from the config file).